### PR TITLE
Feature - submit application page api

### LINKED ISSUE #3333: Connect Submit Application page to API
# Description

When Submit button is pressed on the Step 2 of the application, call the Submit Application mutation #3332. If the submission is successful navigate to the My Applications page.

https://www.figma.com/proto/N9NoPunkqaynjsi89Wrbh0/Direct-Intake---Poster-and-application?scaling=min-zoom&page-id=0%3A1&starting-point-node-id=226%3A4567&node-id=229%3A5735

# Acceptance Criteria
- [ ] When clicking submit, the Submit Application mutation is called
- [ ] A toast shows whether submission was successful or failed
- [ ] If successful, user is navigated to the My Applications page (which shows all their applications with statuses - need to be worked on sep. currently this is a blank page. )
- [ ] A cypress test for this

### PR DESCRIPTION
**Pull request by yonikid15**
Resolves #3333 

## Notes
- [x] When clicking submit, the Submit Application mutation is called
- [x] A toast shows whether submission was successful or failed
- [x] If successful, user is navigated to the My Applications page (which shows all their applications with statuses - need to be worked on sep. currently this is a blank page. )

**UPDATE** these will be handled by issue #3873
- [/] **NEW** Create a new test applicant with a completed profile and a completed application. 
- [/] A cypress test for this (coming soon. still stuck on this unfortunately)

## Testing
- Sign in as `applicant@test.com`
- Go to path `en/browse/applications/{poolCandidateId}/submit`. You will need to grab a poolCandidateId with a `userId` that matches the test applicant (`email: applicant@test.com`).
- The submit button may be disabled if the application is not completed yet. If it is disabled it could be for two reasons: the profile is incomplete OR they don't have all the required skills. You may need to manipulate the applicant profile and/or pools data to change that. I made a new criteria to create a new test applicant with a completed profile and a completed application. This should help with testing. 

(Technically they wouldn't be able to make it to step 2 unless the application was complete.).

### CODE REVIEW

File: frontend/common/src/components/skills/MissingSkills/MissingSkills.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export interface MissingSkillsProps {
   addedSkills?: Skill[];
 }
 
-const getMissingSkills = (required: Skill[], added?: Skill[]) => {
+export const getMissingSkills = (required: Skill[], added?: Skill[]) => {
```
Comment (esizer) [inline frontend/common/src/components/skills/MissingSkills/MissingSkills.tsx:14]: Would it be a good idea to move this to `frontend/common/src/helpers` if we are going to be using it in different places?

File: frontend/talentsearch/src/js/components/experienceAndSkills/ExperienceAndSkills.tsx [hunk 2]
```diff
@@ -68,7 +68,9 @@ export const compareByDate = (e1: ExperienceForDate, e2: ExperienceForDate) => {
   return e2EndDate - e1EndDate;
 };
 
-const flattenExperienceSkills = (experiences: MergedExperiences): Skill[] => {
+export const flattenExperienceSkills = (
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/experienceAndSkills/ExperienceAndSkills.tsx:71]: Same as `getMissingSkills`, I can see moving this to helpers as beneficial.

**Review by esizer — CHANGES_REQUESTED** [review]
I know you are working on some tests still but this is looking great so far! Just noticed some lint warnings and left two small suggestions about the utility functions.

File: frontend/common/src/components/skills/MissingSkills/MissingSkills.tsx (hunk 1 above)
Comment (yonikid15) [reply frontend/common/src/components/skills/MissingSkills/MissingSkills.tsx]: Fixed by [Move reusable functions into helper folders](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3863/commits/3aae2129d3f34a9a21bcf9b33d6a421e3e73480d)

File: frontend/talentsearch/src/js/components/experienceAndSkills/ExperienceAndSkills.tsx (hunk 2 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/experienceAndSkills/ExperienceAndSkills.tsx]: Fixed by [Move reusable functions into helper folders](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3863/commits/3aae2129d3f34a9a21bcf9b33d6a421e3e73480d)

**Review by esizer — APPROVED** [review]
Awesome! 😁

**Pull request merged**